xbmc/interfaces/legacy/AddonUtils.h

```diff
@@ -68,6 +69,9 @@ namespace XBMCAddonUtils
   };
 
 #define LOCKGUI XBMCAddonUtils::GuiLock __gl
+#define LOCKGUIIF(cond) std::unique_ptr<XBMCAddonUtils::GuiLock> __gl; \
```

Comment:
it was 2 lines of code that had to be pasted all over. macro use was a no brainer. the whole point of this macro IS to lock the GIL. we have an on-screen listitem and it needs to not be accessed by any thread in the the python interpreter while we update it on the c++ side.